flamingpy/codes/graphs/egraph.py

```diff
@@ -214,3 +214,111 @@ def draw_adj(self, **kwargs):
 
         adj = self.adj_generator(sparse=False)
         return plot_mat_heat_map(adj, **kwargs)
+
+    def add_qubit(self, qubit=None, index=None, neighbors=None) -> None:
+        """Add qubit to EGraph at position with neigbours in
+        existing_neighbours.
+
+        Args:
+            qubit (3D tuple, or None): qubit to add. If it is a 3D tuple,
+            the qubit is added in that position. If qubit is None, the qubit is
+            positioned one unit further than the maximum position in the z
+            direction in position (0,0,z_max + 1).
+
+            index (int or None): index of qubit. If int, the qubit is
+            positioned at that index. If None, the index of the qubit is the
+            last one.
```

Comment:
We should not let the user potentially replace an existing index by accident. Because we don't have index customizability in index_generator (yet), we should not have it here.

Reply:
Sounds good ✅